backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/menu/context/SysONExplorerTreeItemContextMenuEntryProvider.java

```diff
@@ -51,5 +53,13 @@ public boolean canHandle(IEditingContext editingContext, TreeDescription treeDes
         return tree.getId().startsWith(ExplorerDescriptionProvider.PREFIX)
                 && Objects.equals(tree.getDescriptionId(), this.sysONTreeViewDescriptionProvider.getDescriptionId());
     }
+    
+    @Override
+    public List<ITreeItemContextMenuEntry> getTreeItemContextMenuEntries(IEditingContext editingContext, TreeDescription treeDescription, Tree tree, TreeItem treeItem) {
+        // Remove the expandAll entry: it is handled at the view level in SysON
+        return super.getTreeItemContextMenuEntries(editingContext, treeDescription, tree, treeItem).stream()
+                .filter(entry -> !Objects.equals(entry.id(), "expandAll"))
+                .toList();
+    }
```

Comment:
This is not clean, but we have two options:

- extend `ExplorerTreeItemContextMenuEntryProvider` and filter out the things we don't want
- do not extend anything but we have to duplicate the code for the entries we still want (in particular the "update library" one).

The second option duplicates quite a lot of code, so I went for the more maintainable one.
In the future (not in this contribution), I think we should define all the entries at the view level. SysON explorer is defined at the view level, it is a bit strange to have some entries defined at the view level and some others in this provider.

For the record, this provider is used in Sirius Web to contribute entries to contextual menus of the explorer because the explorer is **not** defined at the view level.

Reply:
ok